### PR TITLE
Feature/multiselect

### PR DESCRIPTION
**Pull request by dgibson666**
* First cut of multi-select dropdown
* Updates to dropdown, dropdown-item, and checkbox to work with multi-select pattern
* Added placeholder functionality to single select dropdowns (leveraged in multi-select as well)
* Minor updates to Icon and docs

I think a few of my styles for the counter/clear button may still be off, but I have a couple things to discuss with Jeremy as well. Please review in the meantime anyway. I'm tired of looking at this and may have missed something in my cleanup efforts.

### CODE REVIEW

File: packages/web-components/src/components/cbp-dropdown-item/cbp-dropdown-item.tsx [hunk 1]
```diff
@@ -19,29 +22,49 @@ export class CbpDropdownItem {
...
-    this.dropdownItemClick.emit({
-      host: this.host,
-      label: label,
-      value: (this.value) ? this.value : label
-    });
-    this.selected=true;
+  handleClick({target}) {
+    // Ignore a click on the label because it will fire a click on the input as well
+    if(target.tagName != "LABEL") {
+      console.log({target});
```
Comment (bagrub) [inline packages/web-components/src/components/cbp-dropdown-item/cbp-dropdown-item.tsx]: do we want this console log pushed?

File: packages/web-components/src/components/cbp-dropdown/cbp-dropdown.tsx [hunk 2]
```diff
@@ -52,36 +56,151 @@ export class CbpDropdown {
...
+    if(this.multiple) {
+      // TechDebt: this should ideally be async/promise. Update: Made selectedItems a State, so this might be fine now. Needs testing.
+      host.selected=!host.selected; // toggle the selected state of the item
+
+      setTimeout( () => {
+        this.selectedItems=Array.from(this.host.querySelectorAll('cbp-dropdown-item[selected]'));
+        this.placeholder=this.selectedItems.length != 1 ? 'Selected Items' : 'Selected Item';
+      }, 50);
+
+      this.value = []; // TODO: make an array of the values of selected items
```
Comment (bagrub) [inline packages/web-components/src/components/cbp-dropdown/cbp-dropdown.tsx]: is the todo here part of the tech debt on line 69?

**Review by bagrub — COMMENTED** [review]
some small stuff, just confirming you want to push

File: packages/web-components/src/components/cbp-dropdown/cbp-dropdown.tsx (hunk 2 above)
Comment (dgibson666) [reply packages/web-components/src/components/cbp-dropdown/cbp-dropdown.tsx]: updated and tested. seems to be working.

File: packages/web-components/src/components/cbp-dropdown-item/cbp-dropdown-item.tsx (hunk 1 above)
Comment (dgibson666) [reply packages/web-components/src/components/cbp-dropdown-item/cbp-dropdown-item.tsx]: removed

File: packages/web-components/src/components/cbp-dropdown/cbp-dropdown.tsx (hunk 2 above)
Comment (dgibson666) [reply packages/web-components/src/components/cbp-dropdown/cbp-dropdown.tsx]: updated and tested.

**Review by bagrub — COMMENTED** [review]
lgtm

**Pull request merged**